src/hdnode.js

```
@@ -194,29 +192,23 @@ HDNode.prototype.toBase58 = function (__isPrivate) {
 // https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#child-key-derivation-ckd-functions
 HDNode.prototype.derive = function (index) {
   var isHardened = index >= HDNode.HIGHEST_BIT
-  var indexBuffer = new Buffer(4)
-  indexBuffer.writeUInt32BE(index, 0)
-
-  var data
+  var data = new Buffer(37)
 
   // Hardened child
   if (isHardened) {
     if (!this.keyPair.d) throw new TypeError('Could not derive hardened child key')
 
     // data = 0x00 || ser256(kpar) || ser32(index)
-    data = Buffer.concat([
-      this.keyPair.d.toBuffer(33),
-      indexBuffer
-    ])
+    data[0] = 0x00
+    this.keyPair.d.toBuffer(32).copy(data, 1)
+    data.writeUInt32BE(index, 33)
 
   // Normal child
   } else {
     // data = serP(point(kpar)) || ser32(index)
     //      = serP(Kpar) || ser32(index)
-    data = Buffer.concat([
-      this.keyPair.getPublicKeyBuffer(),
-      indexBuffer
-    ])
+    this.keyPair.getPublicKeyBuffer().copy(data, 0)
```

The compressed public key is always 33 bytes, unless someone forces it to be `uncompressed`.  Do we need to allow that?
Right now, it will silently pass,  but crop the public key to just the X coordinate.